app/models/sample.rb

```diff
@@ -328,6 +337,62 @@ def initiate_s3_cp
     LogUtil.log_err_and_airbrake("Failed to upload S3 sample '#{name}' (#{id}): #{e}")
   end
 
+  def initiate_basespace_file_transfer
+    # Only continue if the sample status is CREATED and not yet UPLOADED.
+    return unless status == STATUS_CREATED
+
+    Rails.logger.info("HERE #{basespace_dataset_id} #{basespace_access_token}")
+
+    files = files_for_basespace_dataset(basespace_dataset_id, basespace_access_token)
+
+    raise "Could not fetch Basespace files for dataset #{basespace_dataset_id} for sample '#{name}' (#{id})" if files.blank?
+
+    files.each do |file|
+      max_tries = 3
```

Comment:
did you see any cases so far where retries were necessary? it would be good to add a comment about this in the code

Reply:
It's in case there are transient network failures. I now do exponential back-off, as Jonathan suggested above. Will add comment.